### DISCUSSION TITLE
Welcome to ble_monitor Discussions!

### DISCUSSION THREAD

**Discussion by Magalex2x14** [custom-components/ble_monitor#191] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by Ernst79** [comment]
@Magalex2x14 I was wondering why we need the `rounding` option. Lets say we remove it and always round the measurement to the number of decimals set by the user (so `rounding` is always `True`), what would we loose in functionality? Users can specify the number of decimals, even up to 9999 decimals if they want, which is more or less the same as `rounding: False`. With PR #199 they can do this at device level. I don't see the benefit of the `rounding` option.

**Reply by Magalex2x14** [reply]
In fact, I remember that we already asked this question, but at that time it made some sense.
And so I agree, yes, there is no need to keep two options.

**Comment by Ernst79** [comment]
Ok, I’ll look into this next weeks and make a proposal.

**Comment by Ernst79** [comment]
@Magalex2x14 I was working on adding settings to device level and wanted to start with the `batt_entities` option. But I realized that this is in fact already sort of build in in HA. We could decide to remove the `batt_entities` option and just always create the battery sensor. If someone doesn't want the battery sensor, he can easily disable it in the UI.

![image](https://user-images.githubusercontent.com/24375847/103351698-5076ff80-4aa4-11eb-9b98-b740e0cf543c.png)

I thought this is the preferred way of dealing with entities (give them all and let the user disable if he doesn't want them) but I can't find it anymore where I saw this in the docs. 

What do you think?

**Reply by ANS-spb** [reply]
Very lacking the battery sensor for BLE tags :) I'm getting the battery percentage with gattool now and it's inconvenient

**Reply by ANS-spb** [reply]
I mean the device_tracker functionality. Battery percentage reads from uuid 0x2a19 in HEX from my BLE tag

**Reply by Ernst79** [reply]
Hi @ANS-spb What do you exactly want? Battery state for devices that are tracked? 

My guess is that the reporting of the battery level will be different for each device. not sure if there is a general format. What kind of BLE tag do you have and what format does it use? Could you send me an advertisement (e.g. with hcidump)

**Reply by ANS-spb** [reply]
HI @Ernst79 
Yes I'm have a BLE tag (Nut mini) and I'm want see its battery status. 
Currently I'm reading battery status via command gatttool -t random --char-read --uuid 0x2a19 -b "E5:00:FE:B8:DA:74", what returning me hex value of battery charge.
Hcidump output for this command there: [nut mini battery hcidump.txt](https://github.com/custom-components/ble_monitor/files/6847109/nut.mini.battery.hcidump.txt)

**Comment by DenisPryt** [comment]
Hi. Thanks for a great integration! I don't have enough expertise with Bluetooth Low Energy. Does old laptop with bluetooth 2.x can receive BLE packets? Or not?
I tried to add this integration to my HA (without bluetooth 5.x) and I have this errors
2021-05-10 21:29:14 ERROR (Thread-9) [custom_components.ble_monitor] HCIdump thread: Runtime error while sending scan request on hci0: Event loop stopped before Future completed. 2021-05-10 21:30:41 ERROR (Thread-9) [custom_components.ble_monitor] HCIdump thread: Runtime error while stop scan request on hci0: Event loop stopped before Future completed. 
I have researched the pinned issue with same error, but I think my problem is "old" bluetooth. Am I right?

**Reply by DenisPryt** [reply]
@Magalex2x14 just remind about question :)

**Reply by Ernst79** [reply]
Im not sure. Are you able to make a hcidump on this machine?

**Reply by DenisPryt** [reply]
@Ernst79 yes, I can.  What exactly I should run?

**Reply by Ernst79** [reply]
`sudo hcidump --raw hci > dump.txt` should create a file with BLE data, can you run this without any problems?

**Reply by DenisPryt** [reply]
@Ernst79 
```
$ sudo hcidump --raw hci
HCI sniffer - Bluetooth packet analyzer ver 5.55
device: hci0 snap_len: 1500 filter: 0x2
```
And then the program hangs

**Reply by Ernst79** [reply]
Not sure then. You could get yourself a usb dongle instead with Bluetooth 4 or 5.

**Reply by DenisPryt** [reply]
@Ernst79 thanks, I think so too
